src/plugins/orangehrmOpenidAuthenticationPlugin/Controller/OpenIdConnectRedirectController.php

```diff
@@ -18,36 +18,160 @@
+
+        try {
+            $providerId = $this->getAuthUser()->getAttribute(AuthUser::PROVIDER_ID);
+            $authProvider = $this->getSocialMediaAuthenticationService()->getAuthProviderDao()
+                ->getAuthProviderById($providerId);
+            $authProviderExtraDetails = $this->getSocialMediaAuthenticationService()->getAuthProviderDao()
+                ->getAuthProviderDetailsByProviderId($providerId);
 
-        $homePagePath = $this->getHomePageService()->getHomePagePath();
-        return $this->redirect($homePagePath);
+            $oidcClient = $this->getSocialMediaAuthenticationService()->initiateAuthentication(
+                $authProviderExtraDetails,
+                $this->getSocialMediaAuthenticationService()->getScope(),
+                $this->getSocialMediaAuthenticationService()->getRedirectURL()
+            );
+
+            $oidcClient->authenticate();
+
+            $userCredentials = new UserCredential();
+            $userCredentials->setUsername($oidcClient->requestUserInfo('email'));
+            $user = $this->getSocialMediaAuthenticationService()->getOIDCUser($userCredentials);
+
+
+            if (empty($user)) {
+                $this->getAuthUser()->addFlash(
+                    AuthUser::FLASH_LOGIN_ERROR,
+                    [
+                        'error' => AuthenticationException::UNEXPECT_ERROR,
+                        'message' => $this->getI18NHelper()->transBySource('No User Found'),
+                    ]
+                );
+                return new RedirectResponse($loginUrl);
+            }
+
+            if (sizeof($user) != 1) {
+                $this->getAuthUser()->addFlash(
+                    AuthUser::FLASH_LOGIN_ERROR,
+                    [
+                        'error' => AuthenticationException::UNEXPECT_ERROR,
+                        'message' => $this->getI18NHelper()->transBySource('Unexpected Error!'),
+                    ]
+                );
+                return new RedirectResponse($loginUrl);
+            }
+
+            $user = reset($user);
+            if (!($user instanceof User)) {
+                $this->getAuthUser()->addFlash(
+                    AuthUser::FLASH_LOGIN_ERROR,
+                    [
+                        'error' => AuthenticationException::UNEXPECT_ERROR,
+                        'message' => $this->getI18NHelper()->transBySource('Unexpected Error!'),
+                    ]
+                );
+                return new RedirectResponse($loginUrl);
+            }
+
+            if (!$user->getStatus()) {
+                $this->getAuthUser()->addFlash(
+                    AuthUser::FLASH_LOGIN_ERROR,
+                    [
+                        'error' => AuthenticationException::USER_DISABLED,
+                        'message' => $this->getI18NHelper()->transBySource('User Disabled'),
+                    ]
+                );
+            }
```

Comment:
These conditions should goes to service class
See [here](https://github.com/orangehrm/orangehrm/blob/main/src/plugins/orangehrmAuthenticationPlugin/Service/AuthenticationService.php#L40-L56) how to handle exceptions.